app/models/grader.rb

```diff
@@ -21,18 +23,29 @@ def self.unique
     select(column_names - ["id"]).distinct
   end
 
-  def grade(assignment, submission)
+  def really_grade(asn_id, sub_id)
+    puts "XX really grade"
+    puts self.inspect
+
+    assignment = Assignment.find(asn_id)
+    submission = Submission.find(sub_id)
+
     ans = do_grading(assignment, submission)
     submission.compute_grade! if submission.grade_complete?
-    ans
+  end
+
+  def grade(assignment, submission, prio = 0)
```

Comment:
I was wondering about the synchronous calls. Those don't work - running a grading script may take a long time, and we don't want to be blocking a web request on it. Those call sites should be modified to do a field access and accept "not done yet" as an answer.